yang/ietf-external-transaction-id.yang

```diff
@@ -58,46 +56,50 @@ module ietf-external-transaction-id {
     config false;
     description
       "Contains the IDs of configuration transactions that are
-       external to the current device.";
+       external to the device.";
     list configuration-change {
-      key "local-commit-id";
+      key "local-commit-id timestamp";
       description
         "List of configuration changes, identified by their
          local-commit-id";
       leaf local-commit-id {
         type string;
         description
-          "Id as saved by the server. Can be used to retrieve
+          "Stores the identifier as saved by the server. Can be used to retrieve
            the corresponding changes using the server mechanism
            if available.";
       }
-      leaf northbound-transaction-id {
+      leaf timestamp {
```

Comment:
I am assuming that the timestamp, and other metadata associated to the configuration changes would be already stored by the device.  For the NETCONF world, that might be normalized (I didn’t do my homework). On current devices, there is usually a table mapping that local commit id to timestamp, protocol used, user name and so on.

In other words, I would not add to this model the information needed to identify the local-commit-id to trace.

Reply:
If you want to allow for recycling IDs, having timestamps is a clean/simple way to filter/target events and speed up the tracing process.